### ISSUE TITLE
16 bit images?

### ISSUE THREAD

**Issue by ycui123** [fizyr/keras-retinanet#589]
Hi,

Since my images are 16 bit png images, I wonder if this repo supports that? I notice that the code uses pillow to parse image as np.array (but instead of using uint16 array but convert to 8 bits). If it does support, do I need to modify a lot of code? Thanks a lot

**Comment by hgaiser** [comment]
It does not support 16 bit images. You could subclass one of the generators and change the `load_image` (to load as 16 bit images) and `preprocess_image` (to correctly normalize your images) functions to work with your 16 bit images. Are they also one channel images? Because the pretrained weights expect 3 channels. If they're one channel images you can't use the pretrained weights, which makes it more problematic.

As this is not a bug, I'm closing the issue. Further discussion can be held on Slack.

**Comment by ycui123** [comment]
@hgaiser  Thank you for the reply. I have modified the code to make it work for 16bit png images (Use cv2 to process images). And my images have 3 channels so it should be fine.

**Comment by hgaiser** [comment]
Alright, don't forget `preprocess_image` though.
